### PR TITLE
Replace esy with opam

### PR DESCRIPTION
**Pull request by schinns**
Add opam package named melange-react-intl-ppx and remove esy.

### CODE REVIEW

File: test/dune [hunk 1]
```diff
@@ -0,0 +1,4 @@
+(library
```
Comment (jchavarri) [inline test/dune:1]: I think the `bsconfig.json` file can be removed now.

File: bin/dune [hunk 2]
```diff
@@ -1,4 +1,4 @@
 (executable
- (name BsReactIntlPpxBin)
- (public_name bs-react-intl-ppx)
- (libraries bs-react-intl-ppx.lib ppxlib))
+ (name MelangeReactIntlPpxBin)
+ (public_name melange-react-intl-ppx)
```
Comment (jchavarri) [inline bin/dune:3]: I am not sure this exe should be public, I would only make the ppx library public.

If it has to be public, it should prob need to have a different name than the library.

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	opam switch create . 4.14.1 -y --deps-only
+
+.PHONY: init
+init: create-switch install ## Configure everything to develop this repository in local
+
+.PHONY: install
+install: ## Install development dependencies
+	opam update
+	opam install -y . --deps-only
+# opam pin -y add $(project_name).dev . --working-dir
```
Comment (ixzzd) [inline Makefile:25]: Do we need this one?

File: test/dune [hunk 4]
```diff
@@ -0,0 +1,4 @@
...
+ (name Test)
+ (modes melange)
+ (libraries bs_react_intl_ppx))
```
Comment (jchavarri) [inline test/dune:4]: This should be `melange-react-intl-ppx` right? or `MelangeReactIntlPpx` (I am not sure how this was working before)

File: melange-react-intl-ppx.opam [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  "dune" {>= "3.0"}
+  "ppxlib" {>= "0.21.0"}
+  "reason" {>= "3.7.0"}
+  "re" {= "1.10.4"}
+  "ocaml-lsp-server" {with-dev-setup}
+  "ocamlformat" {with-dev-setup}
+  "melange" {with-dev-setup}
+]
+
+build: [["dune" "build" "-p" "bs-react-intl-ppx"]]
```
Comment (jchavarri) [inline melange-react-intl-ppx.opam:23]: Should be 
```suggestion
build: [["dune" "build" "-p" "melange-react-intl-ppx"]]
```

**Review by jchavarri — CHANGES_REQUESTED** [review]
Looking good! left some notes, it would also be great if:

- readme was updated to reflect the changes ([example](https://github.com/ahrefs/melange-react-virtualized/pull/1/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5))
- all the `bs-` names were replaced to `melange-` so that we can rename the repo name just after merging this PR

**Comment by ixzzd** [comment]
We also can remove [GH actions workflow](https://github.com/ahrefs/melange-moment/tree/main/.github/workflows), it was used to build binaries

File: test/dune (hunk 4 above)
Comment (schinns) [reply test/dune]: https://github.com/ahrefs/bs-react-intl-ppx/blob/1cea617f9b80a4f417c05ef5657515ba9cc3b60f/test/dune#L4

File: bin/dune (hunk 2 above)
Comment (schinns) [reply bin/dune]: Removed public name:
https://github.com/ahrefs/bs-react-intl-ppx/blob/1cea617f9b80a4f417c05ef5657515ba9cc3b60f/bin/dune#L1-L3

**Comment by schinns** [comment]
Tested:
```
$ opam pin add melange-react-intl-ppx.dev git+https://github.com/ahrefs/bs-react-intl-ppx.git#eb36bab7565552fffcc59917913220f52c643db8
[melange-react-intl-ppx.dev] synchronised (git+https://github.com/ahrefs/bs-react-intl-ppx.git#eb36bab7565552fffcc59917913220f52c643db8)
melange-react-intl-ppx is now pinned to git+https://github.com/ahrefs/bs-react-intl-ppx.git#eb36bab7565552fffcc59917913220f52c643db8 (version dev)

The following actions will be performed:
  ↗ upgrade melange-react-intl-ppx 0.1.1+20230626 to dev*
Do you want to continue? [Y/n] Y

<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
⬇ retrieved melange-react-intl-ppx.dev  (no changes)
⊘ removed   melange-react-intl-ppx.0.1.1+20230626
∗ installed melange-react-intl-ppx.dev
Done.
```

File: README.md [hunk 6]
```diff
@@ -1,19 +1,19 @@
...
-Get the package:
+Install [opam](https://opam.ocaml.org/) package manager.
+Then:
 
 ```shell
-# yarn
-yarn add --dev @ahrefs/bs-react-intl-ppx
-# or npm
-npm install --save-dev @ahrefs/bs-react-intl-ppx
+opam pin add melange-react-intl-ppx.dev git+https://github.com/ahrefs/bs-react-intl-ppx.git#eb36bab7565552fffcc59917913220f52c643db8
```
Comment (schinns) [inline README.md:10]:
```suggestion
opam pin add melange-react-intl-ppx.dev git+https://github.com/ahrefs/melange-react-intl-ppx.git#master
```
Should update this line after PR is merged and repo is renamed

File: package.json [hunk 7]
```diff
@@ -1,25 +0,0 @@
-{
-  "name": "@ahrefs/bs-react-intl-ppx",
-  "version": "0.1.1",
```
Comment (schinns) [inline package.json:3]: Where should versions be tracked?

Comment (ixzzd) [reply package.json:3]: https://github.com/search?q=repo%3Aahrefs%2Fbs-react-intl-ppx%200.1.1&type=code

The files where it was mentioned were removed (but HISTORY.md should be update)

Comment (jchavarri) [reply package.json:3]: In opam, versions are added when opening the PR against opam repository, e.g. https://github.com/ocaml/opam-repository/pull/23850/files#diff-6bf6c93a7f51115b310f349270eb5f98a80e15624051ac01ba3ba495083f0807R1 (see the opam file name is `melange.1.0.0/opam`).

> but HISTORY.md should be update

👍 yeah I'd add a note there under "Unreleased" section to indicate when the migration to opam & melange happened.

File: README.md [hunk 8]
```diff
@@ -1,19 +1,19 @@
-# bs-react-intl-ppx
+# melange-react-intl-ppx
+
+These are the [Melange](https://melange.re/) bindings for [react-intl](https://github.com/formatjs/formatjs/tree/main/packages/react-intl).
```
Comment (jchavarri) [inline README.md:3]: Aren't the bindings here? https://github.com/cca-io/rescript-react-intl.

```suggestion
This is a [Melange](https://melange.re/) [preprocessor](https://ocaml.org/docs/metaprogramming) that eases the creation of messages when using [rescript-react-intl](https://github.com/cca-io/rescript-react-intl).
```

Comment (jchavarri) [reply README.md:3]: Should the bindings and the ppx leave under the same repo, and probably the same package? cc @ixzzd

File: README.md [hunk 9]
```diff
@@ -1,19 +1,19 @@
...
 ```shell
-# yarn
-yarn add --dev @ahrefs/bs-react-intl-ppx
-# or npm
-npm install --save-dev @ahrefs/bs-react-intl-ppx
+opam pin add melange-react-intl-ppx.dev git+https://github.com/ahrefs/bs-react-intl-ppx.git#eb36bab7565552fffcc59917913220f52c643db8
 ```
 
-Then add it to `bsconfig.json`:
+Then `react-intl` npm package should also be installed:
```
Comment (jchavarri) [inline README.md:13]: I don't think the ppx needs any npm packages to work.

File: melange-react-intl-ppx.opam [hunk 10]
```diff
@@ -0,0 +1,24 @@
+opam-version: "2.0"
+version: "git"
+maintainer: "Ahrefs"
+authors: "Ahrefs"
+dev-repo: "git+https://github.com/ahrefs/bs-react-intl-ppx.git"
+license: "MIT"
+homepage: "https://github.com/ahrefs/bs-react-intl-ppx"
+bug-reports: "https://github.com/ahrefs/bs-react-intl-ppx/issues"
```
Comment (jchavarri) [inline melange-react-intl-ppx.opam:8]: I'd probably point these to `melange-` already, so there are less things we have to remember to update later on.

**Review by jchavarri — APPROVED** [review]
Looks good! Just some minor suggestions re: readme.

File: README.md (hunk 8 above)
Comment (schinns) [reply README.md]: Good catch. Fixed README:
https://github.com/ahrefs/bs-react-intl-ppx/blob/9e7418251f817c0a44d47820f923285702de54dd/README.md?plain=1#L3

File: README.md (hunk 9 above)
Comment (schinns) [reply README.md]: Good point. Removed this line: https://github.com/ahrefs/bs-react-intl-ppx/pull/12/commits/9e7418251f817c0a44d47820f923285702de54dd

File: melange-react-intl-ppx.opam (hunk 10 above)
Comment (schinns) [reply melange-react-intl-ppx.opam]: Good point. Updated:
https://github.com/ahrefs/bs-react-intl-ppx/blob/9e7418251f817c0a44d47820f923285702de54dd/melange-react-intl-ppx.opam#L7-L8

File: README.md (hunk 6 above)
Comment (schinns) [reply README.md]: https://github.com/ahrefs/bs-react-intl-ppx/blob/9e7418251f817c0a44d47820f923285702de54dd/README.md?plain=1#L10

Updated in advance.

File: package.json (hunk 7 above)
Comment (schinns) [reply package.json:3]: Ah, TIL! Thanks.

Updated HISTORY.md:
https://github.com/ahrefs/bs-react-intl-ppx/blob/9e7418251f817c0a44d47820f923285702de54dd/HISTORY.md?plain=1#L3-L5

**Comment by jchavarri** [comment]
thanks @schinns ! @ixzzd is it good to merge?

**Review by ixzzd — APPROVED** [review]
Thank you 🚀

**Pull request merged**